geps/gep-1619.md

```diff
@@ -157,49 +281,401 @@ Generally, the implementation API programs the dataplane API; however these two
 
 ### Sessions in Java
 
-Java application servers such as Tomcat and Jetty, were the first to standardize the API around persistent sessions. These Java applications introduced the “jsessionid” cookie as well as more advanced features such as session migration, replication, and on demand session activation. It’s important for Gateway API to examine persistent session use cases and history from Java APIs to ensure the API is designed appropriately.
-
-## API
-
-### Session Persistence API
+Java application servers such as Tomcat and Jetty, were the first to standardize the API around cookies and sessions.
+These Java applications introduced the “jsessionid” cookie as well as more advanced features such as session migration,
+replication, and on demand session activation. It’s important for Gateway API to examine cookie use cases and history
+from Java APIs to ensure the API is designed appropriately.
 
-The majority of implementations and dataplanes support configuring cookies for session persistence while support for header-based session persistence is limited.
+### Session Affinity in K8S Services
 
-To organize potential Gateway API configuration of cookie functionality, the next section will break down configuration into specific categories:
-- Session Cookie Configuration
-- Session Cookie Strategy
+Kubernetes provides an API that allows you to enable [session affinity](https://kubernetes.io/docs/reference/networking/virtual-ips/#session-affinity)
+on service objects. It ensures consistent sessions by utilizing the client's IP address and also offers the option to
+set a timeout for the maximum session duration. Implementations of Gateway API don't rely on the service IP, so this
+particular application of session affinity doesn't impact the design of the GEP. However, we recommend that users enable
+session affinity using the Kubernetes service object to for applications requiring persistence, particularly due to the
```

Comment:
My point was that common use of the words - for a long time - has been synonyms. Redefining a word is confusing for users, but maybe if we qualify it "K8S Affinity" and "K8S Persistence" it would be clear. Plenty of products provide "K8S Persistent session" features and are calling it "Affinity".

My understanding of K8S Service affinity is that it meets the definition of K8S Persistence. How it is implemented
is not important - in the absence of NATs and gateways you can implement K8S persistence using client IP ( it is also possible to implement K8S persistence using the user ID - and quite common !). It is not impacted by auto-scaling events, it is truly sticky. If I missed something in our definition of K8S persistence that is not met by K8S Service - let me know.  The client IP is provided by the client ( I assume intention is hard to determine...). And it works for TCP - which is pretty important since all cookie persistences are http only. Would an implementation that uses the JWT or mTLS client identity to send traffic to the Pod hosting the user shard not meet 'persistence' definition ? 

My advice is to just stay away from defining affinity - we have a GEP for session persistence, we can provide a definition for what we support ( requests from one client go to the same backend, as long as it is alive and it didn't expire ). I don't think it's required that we define other words - in particular since K8S v1 Service has a definition for their affinity.
Would be strange/confusing for K8s v1 Service to define affinity in different terms/meaning from K8S Gateway.

We do need to decide if K8S v1 Service meets the definition of persistence - but even if it doesn't, a service designed for persistence can benefit from using K8S Service affinity when called directly, and we may want to mention this.